lib/worker/form.js

```diff
@@ -42,5 +42,23 @@ const create = pushDraftToEnketo;
 const updateDraftSet = pushDraftToEnketo;
 const updatePublish = pushFormToEnketo;
 
-module.exports = { create, updateDraftSet, updatePublish };
+const updateEntitiesVersion = async ({ Forms }, event) => {
+  const { projectId, xmlFormId } = await Forms.getByActeeIdForUpdate(event.acteeId).then(o => o.get());
+  const publishedVersion = await Forms.getByProjectAndXmlFormId(projectId, xmlFormId, true, Form.PublishedVersion).then(o => o.orNull());
+  if (publishedVersion && publishedVersion.currentDefId != null) {
```

Comment:
I thought it would work like that, too, but `getByProjectAndXmlFormId()` always returns the base Form frame. It does/doesn't include the FormDef based on the version argument. (See internals of `_getSql`.) If the form is only a draft, `publishedVersion` exists there, but has a null `def` and `currentDefId` is null as well.

Reply:
Ah interesting, that sort of makes me think that the initial null check might not be needed then. If the form didn't exist at all, then the job should have already stopped after `Forms.getByActeeIdForUpdate()`. It sounds like `publishedVersion` will always exist, it just might not have a `def`. Instead of `o.orNull()`, how about `o.get()`? I think that'd signal that we really expect `publishedVersion` to exist here.